packages/ui/src/components/Table/STable.vue

```diff
@@ -0,0 +1,819 @@
+
+function sort({ prop, order }: { prop: string; order: ColumnSortOrder }) {
+  const column = getColumnByProp(prop)
+
+  if (column) {
+    sortExplicitly(column, order)
+  }
+}
+
+defineExpose({
+  /**
+   * used in multiple selection Table, clear user selection
+   * */
+  clearSelection: manualClearSelection,
+  /**
+   * used in multiple selection Table, toggle if a certain row is selected.
+   * With the second parameter, you can directly set if this row is selected
+   * */
+  toggleRowSelection: manualToggleRowSelection,
+  /**
+   * used in multiple selection Table, toggle the selected state of all rows
+   * */
+  toggleAllSelection: () => {
+    if (activeSelectionColumn.value) {
+      manualToggleAllSelection(activeSelectionColumn.value)
+    }
+  },
+  /**
+   * used in expandable Table or tree Table, toggle if a certain row is expanded.
+   * With the second parameter, you can directly set if this row is expanded or collapsed
+   * */
+  toggleRowExpansion: toggleRowExpanded,
+  /**
+   * sort Table manually. Property prop is used to set sort column, property order is used to set sort order
+   * */
+  sort,
+  /**
+   * clear sorting, restore data to the original order
+   * */
+  clearSort,
+  /**
+   * used in single selection Table, set a certain row selected.
+   * If called without any parameter, it will clear selection.
+   * */
+  setCurrentRow,
+})
+
+function getColumnByProp(prop: string) {
+  return columns.filter(isDefaultColumn).find((column) => column.prop === prop)
+}
+
+function getRowKey(row: TableRow) {
+  if (typeof props.rowKey === 'string') {
+    if (props.rowKey.includes('.')) {
+      return row[props.rowKey]
+    }
+
+    let keys = props.rowKey.split('.')
+    let current: unknown = row
+
+    for (let key of keys) {
+      if (isRecord(current) && key in current) {
+        current = current[key]
+      }
+    }
+
+    return current
+  }
+
+  if (typeof props.rowKey === 'function') {
+    return props.rowKey(row)
+  }
+}
+
+function getStyleOrClass<T extends object | string>(prop: T | (() => T), args?: undefined): T | undefined
+function getStyleOrClass<T extends object | string, S>(prop: T | ((args: S) => T), args: S): T | undefined
+function getStyleOrClass<T extends object | string, S>(prop: T | ((args: S) => T), args: S): T | undefined {
+  if (typeof prop === 'function') {
+    return prop(args)
+  }
+
+  return prop || undefined
+}
+
+function isCheckBoxDisabled(column: ActionColumnApi, row: TableRow, index: number) {
+  return !column.selectable || column.selectable(row, index)
+}
+
+function getDefaultCellValue(row: TableRow, column: ColumnApi, index: number) {
+  return column.formatter ? column.formatter(row, column, get(row, column.prop), index) : get(row, column.prop)
+}
+
+function getCellTooltipContent(row: TableRow, column: ColumnApi | ActionColumnApi) {
+  if (!column.showOverflowTooltip || !isDefaultColumn(column)) {
+    return
+  }
+
+  return String(get(row, column.prop))
+}
```

Comment:
These functions are pure and don't touch anything in the `setup` context. Maybe move it into `utils`?

It should make compiled setup function a bit lighter, but i don't know, may be Vue Compiler is smart enough to move such stuff out.

Reply:
This methods are too close to table template, so I don't think it is necessary. And I don't understand how moving out them can make `setup` lighter